In [1]:
import pandas as pd
import numpy as np

# Exploratory Data Analysis

In [8]:
df_2000 = pd.read_csv('Data/final_tmdb_data_2000.csv.gz')
df_2001 = pd.read_csv('Data/final_tmdb_data_2001.csv.gz')

In [11]:
df = pd.concat((df_2000, df_2001), axis = 0)

- How many movies had at least some valid financial information (values > 0 for budget OR revenue)?

In [13]:
bud = df['budget'] > 0 
rev = df['revenue'] > 0
df = df.loc[bud | rev]

In [14]:
df.dropna(subset=['budget'], inplace=True)
df.dropna(subset=['revenue'], inplace=True)

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 627 entries, 1 to 1275
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                627 non-null    object 
 1   adult                  627 non-null    float64
 2   backdrop_path          531 non-null    object 
 3   belongs_to_collection  105 non-null    object 
 4   budget                 627 non-null    float64
 5   genres                 627 non-null    object 
 6   homepage               77 non-null     object 
 7   id                     627 non-null    float64
 8   original_language      627 non-null    object 
 9   original_title         627 non-null    object 
 10  overview               625 non-null    object 
 11  popularity             627 non-null    float64
 12  poster_path            606 non-null    object 
 13  production_companies   627 non-null    object 
 14  production_countries   627 non-null    object 
 15  relea

 627 values for revenue were not null, Budget also had 627 non null values.

- How many movies are there in each of the certification categories (G/PG/PG-13/R)?

In [16]:
df['certification'].value_counts()

R        229
PG-13    131
PG        35
NR        17
G         14
Name: certification, dtype: int64

R: 229, PG-13: 131, PG: 35, NR: 17, G: 14

- What is the average revenue per certification category?

In [17]:
df.groupby(['certification']).mean('revenue')

adult        budget            id  popularity       revenue  \
certification                                                                
G                0.0  4.085714e+07  17450.500000   26.335286  1.237463e+08   
NR               0.0  6.302358e+06  73899.705882    9.900118  8.887611e+06   
PG               0.0  4.496429e+07  13296.685714   28.747657  1.126496e+08   
PG-13            0.0  4.315388e+07  15623.366412   15.043687  9.926299e+07   
R                0.0  1.974674e+07  25745.519651   12.132900  3.321119e+07   

                  runtime  video  vote_average   vote_count  
certification                                                
G               85.928571    0.0      6.443857  2496.785714  
NR              99.705882    0.0      5.972471   458.058824  
PG             105.885714    0.0      6.172257  2553.257143  
PG-13          108.908397    0.0      6.091344  1771.206107  
R              105.746725    0.0      5.950821   978.196507

- What is the average budget per certification category?

In [18]:
df.groupby(['certification']).mean('budget')

adult        budget            id  popularity       revenue  \
certification                                                                
G                0.0  4.085714e+07  17450.500000   26.335286  1.237463e+08   
NR               0.0  6.302358e+06  73899.705882    9.900118  8.887611e+06   
PG               0.0  4.496429e+07  13296.685714   28.747657  1.126496e+08   
PG-13            0.0  4.315388e+07  15623.366412   15.043687  9.926299e+07   
R                0.0  1.974674e+07  25745.519651   12.132900  3.321119e+07   

                  runtime  video  vote_average   vote_count  
certification                                                
G               85.928571    0.0      6.443857  2496.785714  
NR              99.705882    0.0      5.972471   458.058824  
PG             105.885714    0.0      6.172257  2553.257143  
PG-13          108.908397    0.0      6.091344  1771.206107  
R              105.746725    0.0      5.950821   978.196507

In [19]:
df.to_csv("Data/tmdb_results_combined.csv.gz",compression='gzip',index=False)